src/index.ts

```diff
@@ -118,13 +118,18 @@ export function project<T = any>(options?: string | Options) {
 /**
  * Determine if the metadata server is currently available.
  */
+let cachedIsAvailableResponse: boolean | undefined;
 export async function isAvailable() {
+  if (cachedIsAvailableResponse !== undefined) {
```

Comment:
We should queue calls when is available cache is undefined but an isAvailable check is pending

Reply:
that's a good idea, couldn't we just hold on to the first promise that's enqueued, and throw it out as soon as we get a success or failure? I can't think of a situation where we wouldn't want to share the `isAvailable` response.